src/python/gudhi/simplex_tree.pxd

```diff
@@ -81,8 +81,8 @@ cdef extern from "Simplex_tree_interface.h" namespace "Gudhi":
         Simplex_tree_skeleton_iterator get_skeleton_iterator_end(int dimension) nogil
         pair[Simplex_tree_boundary_iterator, Simplex_tree_boundary_iterator] get_boundary_iterators(vector[int] simplex) nogil except +
         # Expansion with blockers
-        ctypedef bool (*blocker_func_t)(vector[int], void *user_data)
-        void expansion_with_blockers_callback(int dimension, blocker_func_t user_func, void *user_data)
+        ctypedef bool (*blocker_func_t)(vector[int], void *user_data) except +
+        void expansion_with_blockers_callback(int dimension, blocker_func_t user_func, void *user_data) except +
```

Comment:
I wonder about the exceptions. I don't think the C++ code is likely to throw an exception. If an exception is thrown, it probably comes from the python callback. Would `except *` or `except +*` also work? What happens if we throw an exception from the callback?

Reply:
I think there would also be possibilities adding noexcept on the other side, but this is ok, this function is slow anyway. I filed #935 for the future.